src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        for (User registeredUser : Storage.people) {
+            if (registeredUser.getLogin().equals(user.getLogin())) {
+                throw new RegistrationException(
+                        "User with login " + user.getLogin() + " already exist");
+            }
+        }
```

Comment:
The logic for checking if a user with the same login already exists is correct. However, ensure that this check is performed after null checks for the login to prevent potential `NullPointerException`.